### ISSUE TITLE
Insert/Reserve space before generic arrow type arguments

### ISSUE THREAD

**Issue by eternalphane** [prettier/prettier#4818]
**Prettier 1.13.7**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLWcBOAzAhmOAAgH0BZCAEzgBtC4APeKCgZ0IEkMd84AecqtQB8hYAB0ohKSXSoYqXNVQAvLCySEAgpky4Anr0K8AKnUYJWhATSEAKFjWwbjASkIBeEQDcIqCkIBuCQBfEAAaEAgAB3loFmRQXB0IAHcABSSEeJRcHz9wkAAjXTAAazgYAGUo-HQAc2QYTABXOAj0B0wYNN06gFtcZDxqBwiAKxZ6ACES8qrcPrgAGXQ4IcVRkBrMTuQQGD0ouBYwTFQYgqiz2AB1PxgAC2QADgAGCKuIBxvdKL2r45YLxrCKYOAAR2aqDBPVw-UGSGGmwcfVQjRabRALHq1DgAEVmhB4OsRpiYLhCncKI9kAAWCJNXCoJRQOoAYQgfQGeyg0BBIGaDmMFOySLgwWCQA)
```sh
--parser typescript
--tab-width 4
```

**Input:**
```tsx
interface _Model extends Interface<Model> {
    _initializers: Array< <T extends Model>(self: T) => void>;
}
```

**Output:**
```tsx
interface _Model extends Interface<Model> {
    _initializers: Array<<T extends Model>(self: T) => void>;
//                      ^
//                      error TS1005: ';' expected.
}

```

**Expected behavior:**
same as input.

Related: Microsoft/Typescript#23996

**Comment by ikatyang** [comment]
I'm not sure what should we do here since it's a bug in TS, but you can use [`// prettier-ignore`](https://prettier.io/docs/en/ignore.html#javascript) as a temporary workaround for now:

**Prettier 1.13.7**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLWcBOAzAhmOAAgH0BZCAEzgBtC4APeKCgZ0IEkMd84AecqtQB8hYAB0ohKYQD0MwgAdMcGDFRYAtKgDmUCMonSS6VGtzVUALywskhAIKZMuAJ69CvACp1GCVoQEaIQAKFhpsO08ASkIAXhEANwhUCiEAbgkAXxAAGhAIBTVoFmRQXCcIAHcABXKEEpRcJJTckAAjZzAAaxUAZQV8dG1kGEwAVzg89DDMGGrnbQBbXGQ8ajC8gCsWegAhTp6YXtxFuAAZdDhV8w2QAcwZ5BAYFwU4FjBMVELWpXQYADqKRgAAtkAAOAAMeSUEDCAOcCieSneWASVzyygAjmNUMp5rgliskGtbmFFqgRuNJiAWENqHAAIpjCDwa7rGkwXBtIEUUHIAAseVGuFQFig2gAwhBFssnnooBiQGMwp5uQ1SXBMpkgA)
```sh
--parser typescript
--tab-width 4
```

**Input:**
```tsx
interface _Model extends Interface<Model> {
    // prettier-ignore
    _initializers: Array< <T extends Model>(self: T) => void>;
}
```

**Output:**
```tsx
interface _Model extends Interface<Model> {
    // prettier-ignore
    _initializers: Array< <T extends Model>(self: T) => void>;
}

```

**Comment by thorn0** [comment]
The TS bug was fixed in TypeScript 3.3.
